### PR TITLE
fix: correct VM RAM allocation check for Windows and macOS. close #1191

### LINKED ISSUE #1191: bug: RAM allocation for macOS and Windows VMs can prevent VMs from booting
# Expected behavior

running "./quickemu --vm windows-10.conf --display spice" should run windows VM

## Actual behavior

running "./quickemu --vm windows-10.conf --display spice" show error:

Quickemu 4.9.4 using /usr/bin/qemu-system-x86_64 v8.2.2
 - Host:     Fedora Linux 40 (Workstation Edition) running Linux 6.8 (smithix)
 - CPU:      Intel(R) Core(TM) i7-4700MQ CPU @ 2.40GHz
 - CPU VM:   1 Socket(s), 2 Core(s), 2 Thread(s), 4G RAM
ERROR! You have insufficient RAM to run windows in a VM

# Steps to reproduce the behavior

./quickemu --vm windows-10.conf --display spice

# Additional context
I thing in this code:

    if [ "${guest_os}" == "macos" ] || [ "${guest_os}" == "windows" ] || [ "${guest_os}" == "windows-server" ]; then
        if [ "${RAM_VM//G/}" -lt 8 ]; then
            echo "ERROR! You have insufficient RAM to run ${guest_os} in a VM"
            exit 1
        fi
    fi

RAM_VM should be RAM_HOST

## System Information
----------------------------------
        Quickemu 4.9.4
----------------------------------
Distro:	Fedora Linux 40 (Workstation Edition)
Kernel:	Linux 6.8.8-300.fc40.x86_64 x86_64
Memory:	16G
GPU:
 - Intel Corporation 4th Gen Core Processor Integrated Graphics Controller (rev 06)

### PR DESCRIPTION
**Pull request by flexiondotorg**

### CODE REVIEW

**Review by bennydente — APPROVED** [review]

File: quickemu [hunk 1]
```diff
@@ -330,9 +330,16 @@ function vm_boot() {
     fi
     echo ", ${RAM_VM} RAM"
 
-    if [ "${guest_os}" == "macos" ] || [ "${guest_os}" == "windows" ] || [ "${guest_os}" == "windows-server" ]; then
+    if [ "${guest_os}" == "windows" ] || [ "${guest_os}" == "windows-server" ]; then
+        if [ "${RAM_VM//G/}" -lt 4 ]; then
+            echo "ERROR! The guest virtual machine has been allocated insufficient RAM to run Windows."
+            echo "       You can override the guest RAM allocation by adding 'ram=4G' to ${VM}"
```
Comment (bennydente) [inline quickemu:336]: mybe "You can override the guest RAM allocation by adding 'ram=4G' to ${VM} 's config file"

File: quickemu [hunk 2]
```diff
@@ -330,9 +330,16 @@ function vm_boot() {
...
+            exit 1
+        fi
+    elif [ "${guest_os}" == "macos" ]; then
         if [ "${RAM_VM//G/}" -lt 8 ]; then
-            echo "ERROR! You have insufficient RAM to run ${guest_os} in a VM"
+            echo "ERROR! The guest virtual machine has been allocated insufficient RAM to run macOS."
+            echo "       You can override the guest RAM allocation by adding 'ram=8G' to ${VM}"
```
Comment (bennydente) [inline quickemu:342]: mybe "You can override the guest RAM allocation by adding 'ram=4G' to ${VM} 's config file"

**Review by bennydente — APPROVED** [review]

**Review by bennydente — APPROVED** [review]

File: quickemu (hunk 1 above)
Comment (flexiondotorg) [reply quickemu:336]: `${VM}` expands to the name of the config file.

File: quickemu (hunk 2 above)
Comment (flexiondotorg) [reply quickemu:342]: `${VM}` expands to the name of the config file.

**Pull request merged**